homeassistant/components/ipp/config_flow.py

```diff
@@ -99,25 +100,27 @@ async def async_step_user(
 
         return self.async_create_entry(title=user_input[CONF_HOST], data=user_input)
 
-    async def async_step_zeroconf(self, discovery_info: dict[str, Any]) -> FlowResult:
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
         """Handle zeroconf discovery."""
-        port = discovery_info[CONF_PORT]
-        zctype = discovery_info["type"]
-        name = discovery_info[CONF_NAME].replace(f".{zctype}", "")
+        port = discovery_info[zeroconf.ATTR_PORT]
+        zctype = discovery_info[zeroconf.ATTR_TYPE]
+        name = discovery_info[zeroconf.ATTR_NAME].replace(f".{zctype}", "")
```

Comment:
It might be nice to provide a utility function in the future to strip the type since a lot of integrations need to do it. `.replace` isn't the best solution here but it works due to the allowable format.